Modify release documentation to add detailed instructions on permissions (#2455)
MarkBind/markbind: docs/devGuide/projectManagement.md

    * For GitHub, you need rights to **push to master branch** and **make new releases**.
        * To check if you can make a new release and push to master branch, go to the [release page](https://github.com/MarkBind/markbind/releases) and check for the "Draft a new release" button. 
          If missing, you may not have permissions for a release.

    * For npm, you need to be in the [MarkBind organization](https://www.npmjs.com/org/markbind).
   
      * To check if you are in the MarkBind organization, go to your npm profile and check if MarkBind is listed under organizations.
        <pic src="/images/npm-profile.jpg" width="100%" alt="npm profile">
        Example of profile that has been added to Markbind organisation
        </pic>
      * There should be 4 packages listed under the organization, `markbind-cli`, `@markbind/core`, `@markbind/core-web` and `@markbind/vue_components`. 
      * Notably, the first three are packages that we publish every release while the last one has since become a private package consumed internally.

> 1. **Login to your npm account in your terminal** by running `npm login`

Reviewer: ```suggestion
1. **Login to your npm account in your terminal** by running `npm login`.
```

   </box>

## Doing a Release

> 1. **Make sure you have the correct permissions** for [MarkBind's GitHub repository](https://github.com/MarkBind/markbind) and [npm organization](https://www.npmjs.com/org/markbind)

Reviewer: ```suggestion
1. **Make sure you have the correct permissions** for [MarkBind's GitHub repository](https://github.com/MarkBind/markbind) and [npm organization](https://www.npmjs.com/org/markbind).
```

## Doing a Release

> 1. **Make sure you have the correct permissions** for [MarkBind's repository](https://github.com/MarkBind/markbind) and [npm](https://www.npmjs.com/)

Reviewer: ```suggestion
1. **Make sure you have the correct permissions** for [MarkBind's GitHub repository](https://github.com/MarkBind/markbind) and [npm organization](https://www.npmjs.com/org/markbind)
```

>     * For Markbind, you need rights to **push to master branch** and **make new releases**

Reviewer: ```suggestion
    * For GitHub, you need rights to **push to master branch** and **make new releases**
```

    <box type="tip" seamless>

>     To check your github permissions, follow the instructions [here](https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#get-repository-permissions-for-a-user).

Reviewer: Any simpler way to check? Or could we provide the curl command/link that dev can visit by replacing the param with their username to get the result? Linking the API docs here is nice but would be better if we give a simple copy-paste validation for the future maintainer, wdyt?

Author: There is a simpler way to do this according to [stackoverflow](https://stackoverflow.com/a/40326507) which is just to check if you can edit a file on the repo. I was a bit reluctant to write this here because I'm not sure how legitimate this way is. 
But this does seem a lot easier to just do a quick check like this, rather than have to do a curl.

And I was concerned that the curl command/link would change due to API changes. 

TBH I'm kind of leaning towards using the 'just check if you can edit a file way' because its a lot easier, rather than having to like do a curl, get git hub token and so on.

What do you all think!

Reviewer: I suppose your point on making a release might suffice (not sure)? if so it's a easy way to verify by checking whether the maintainer can make release -> probably have the same rights to push to master

Reviewer 2: Perhaps using a list here could make it slightly more readable in my opinion e.g.
To check:
- Github permissions...
- If you can make a new release....

Author: > I suppose your point on making a release might suffice (not sure)? if so it's a easy way to verify by checking whether the maintainer can make release -> probably have the same rights to push to master

Hm I think you are right @tlylt according to this [documentation](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository)

I'll just remove this check that since if you can do a release - you should be able to push to master.

>     To check if you can make a new release, go to the [release page](https://github.com/MarkBind/markbind/releases) and check for the "Draft a new release button". If missing, you may not have permissions for a release.

Reviewer: good point here!

    </box>

>     * For npm, you need to be in the [Markbind organization](https://www.npmjs.com/org/markbind)

Reviewer: ```suggestion
    * For npm, you need to be in the [MarkBind organization](https://www.npmjs.com/org/markbind)
```

   <box type="tip" seamless>
   

>    To check if you are in the Markbind organization, go your npm profile and check if Markbind is listed under organizations.

Reviewer: ```suggestion
   To check if you are in the MarkBind organization, go to your npm profile and check if MarkBind is listed under organizations.
```

>    There should be 4 packages listed under the organization, `markbind-cli`, `@markbind/core`, `@markbind/core-web` and `@markbind/vue_components`

Reviewer: ```suggestion
   There should be 4 packages listed under the organization, `markbind-cli`, `@markbind/core`, `@markbind/core-web` and `@markbind/vue_components`. Notably, the first three are packages that we publish every release while the last one has since become a private package consumed internally.
```

   
   <pic src="/images/npm-profile.jpg" width="100%" alt="npm profile">
   Example of profile that has been added to Markbind organisation
   </pic>

   </box>

> 1. **Login to npm in your terminal** using `npm login`

Reviewer: ```suggestion
1. **Login to your npm account in your terminal** by running `npm login`
```
